### PR TITLE
Add Alpine package build

### PR DESCRIPTION
**Pull request by okynos**
|Related issue|
|---|
| Related wazuh/wazuh-jenkins#3994 |

## Description
We have added support to Alpine package building

## Tests

- Build the package in any supported platform
  - [x] Alpine
- [x] Package installation
- [x] Package upgrade
- [x] Package remove
- [x] Package install/remove/install

### CODE REVIEW

File: alpine/build_package.sh [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    echo "    -d, --debug                [Optional] Build the binaries with debug symbols. By default: no."
+    echo "    --dont-build-docker        [Optional] Locally built docker image will be used instead of generating a new one."
+    echo "    --sources <path>           [Optional] Absolute path containing wazuh source code. This option will use local source code instead of downloading it from GitHub."
+    echo "    --packages-reference <ref> [Optional] Select Git branch or tag from wazuh-packages repository. e.g ${spec_reference}"
+    echo "    --dev                      [Optional] Use the SPECS files stored in the host instead of downloading them from GitHub."
+    echo "    --future                   [Optional] Build test future package 99.99.0 Used for development purposes."
+    echo "    -h, --help                 Show this help."
+    echo
+    exit $1
+}
```
[the remark this answers is not in the document]
Comment (okynos) [reply alpine/build_package.sh]: Done in b949b7fe

File: alpine/SPECS/wazuh-agent/wazuh-agent.post-deinstall [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+if command -v getent > /dev/null 2>&1 && getent group wazuh > /dev/null 2>&1; then
+  delgroup wazuh > /dev/null 2>&1
+elif id -g wazuh > /dev/null 2>&1; then
+  delgroup wazuh > /dev/null 2>&1
+fi
+
+# Remove lingering folders and files
+rm -rf ${directory_base}
+
+exit 0
```
[the remark this answers is not in the document]
Comment (okynos) [reply alpine/SPECS/wazuh-agent/wazuh-agent.post-deinstall]: Done in b949b7fe

**Review by rauldpm — APPROVED** [review]
LGTM

**Review by alberpilot — APPROVED** [review]
LGTM

**Pull request merged**